### PR TITLE
Add GCR SA for Production

### PR DESCRIPTION
**Pull request by snf2ye**

### CODE REVIEW

File: datarepo/modules/datarepo-app/serviceaccounts.tf [hunk 1]
```diff
@@ -76,3 +76,42 @@ resource "vault_generic_secret" "test_runner_sa_key" {
...
+resource "google_service_account" "datarepo_gcr_sa" {
+  count        = var.environment == production ? 1 : 0
+  provider     = google.target
+  project      = var.google_project
+  account_id   = "${var.service}-${local.owner}-gcr"
+  display_name = "${var.service}-${local.owner}-gcr"
+}
+
+resource "google_project_iam_member" "gcr_sa_role" {
+  count = var.enable ? length(local_gcr_roles) : 0
```
Comment (smark88) [inline datarepo/modules/datarepo-app/serviceaccounts.tf:90]:
```suggestion
  count = var.environment == "production" ? 1 : 0
```

File: datarepo/modules/datarepo-app/serviceaccounts.tf [hunk 2]
```diff
@@ -76,3 +76,42 @@ resource "vault_generic_secret" "test_runner_sa_key" {
...
+
+  provider = google.target
+  project  = var.google_project
+  role     = local.gcr_roles[count.index]
```
Comment (smark88) [inline datarepo/modules/datarepo-app/serviceaccounts.tf:94]:
```suggestion
  role     = "roles/storage.buckets.get"
```

File: datarepo/modules/datarepo-app/serviceaccounts.tf [hunk 3]
```diff
@@ -76,3 +76,42 @@ resource "vault_generic_secret" "test_runner_sa_key" {
 }
 EOT
 }
+
+## gcr-sa
+resource "google_service_account" "datarepo_gcr_sa" {
+  count        = "${var.environment}" == "production" ? 1 : 0
```
Comment (smark88) [inline datarepo/modules/datarepo-app/serviceaccounts.tf:82]:
```suggestion
  count        = var.environment == "production" ? 1 : 0
```

File: datarepo/modules/datarepo-app/serviceaccounts.tf [hunk 4]
```diff
@@ -76,3 +76,42 @@ resource "vault_generic_secret" "test_runner_sa_key" {
...
+
+  provider = google.target
+  project  = var.google_project
+  role     = local.gcr_roles[count.index]
+  member   = "serviceAccount:${google_service_account.datarepo_gcr_sa[0].email}"
+}
+
+## vault write gcr-sa
+resource "google_service_account_key" "gcr_sa_key" {
+  count = "${var.environment}" == "production" ? 1 : 0
```
Comment (smark88) [inline datarepo/modules/datarepo-app/serviceaccounts.tf:100]:
```suggestion
  count = var.environment == "production" ? 1 : 0
```

File: datarepo/modules/datarepo-app/serviceaccounts.tf [hunk 5]
```diff
@@ -76,3 +76,42 @@ resource "vault_generic_secret" "test_runner_sa_key" {
...
+
+  provider           = google.target
+  service_account_id = google_service_account.datarepo_gcr_sa[0].name
+}
+
+resource "vault_generic_secret" "gcr_sa_key" {
+  count = "${var.environment}" == "production" ? 1 : 0
```
Comment (smark88) [inline datarepo/modules/datarepo-app/serviceaccounts.tf:107]:
```suggestion
  count = var.environment == "production" ? 1 : 0
```

File: datarepo/modules/datarepo-app/variables.tf [hunk 6]
```diff
@@ -59,6 +59,12 @@ locals {
   ]
 }
 
+locals {
```
Comment (smark88) [inline datarepo/modules/datarepo-app/variables.tf:62]: remove this local

**Review by fboulnois — APPROVED** [review]
Plan looks good, only change is in prod.

**Review by smark88 — APPROVED** [review]

**Pull request merged**